compiler-core/src/format.rs

```diff
@@ -1083,6 +1093,8 @@ impl<'comments> Formatter<'comments> {
         commented(doc_comments.append(doc).group(), comments)
     }
 
+    // TODO : This takes too many arguments
+    #[allow(clippy::too_many_arguments)]
```

Comment:
Could you fix this to take the CustomType struct rather than each of its fields? Thank you.

Reply:
Got you. Will change the signature to take in a struct instead.